Module rename

bmw opened the pull request:
Fixes #2739 and does everything but the shim for #2647.

bmw commented:
This is basically a slightly smarter version of running `sed -i -e 's/letsencrypt/certbot/g' -e 's/Let's Encrypt/Certbot/g'` on our entire project.

Noteworthy things are:
* Most user facing file paths and directories were not modified (see #2830). The only one I can think of that I did modify was creating `certbot.log` instead of `letsencrypt.log` if we error before setting up logging.
* I changed the author from "Let's Encrypt Project" to "Electronic Frontier Foundation" for all packages. I didn't modify `LICENSE.txt` because legalese is scary.
* I only changed user facing output of `letsencrypt-auto`. There are no `certbot` packages on PyPI yet and we still need to figure out what we're doing with `certbot-auto`.

kuba commented on acme/setup.py:
```diff
@@ -53,7 +53,7 @@
     description='ACME protocol implementation in Python',
     url='https://github.com/letsencrypt/letsencrypt',
-    author="Let's Encrypt Project",
+    author="Electronic Frontier Foundation",
```
that doesn't look correct - previous version captured authors other than EFF. Why not "Certbot Project"? Frankly speaking, it was Let's Encrypt project who authored it so far ;)

bmw replied:
Perhaps. The client was written almost exclusively by you and EFF employees and with the client becoming an EFF project, this seemed fitting to me.

With that said, I don't have too strong of an opinion on the issue. What do you think @pde?

kuba replied:
Well, there were other contributors as well. Shouldn't everybody be captured here (and in other setup.py files)?

SwartzCr replied:
I'm in favor of calling it Certbot Project, since kuba is right that there have been many contributors. I think it makes sense as this is a project (now) managed by EFF, but it is still a freestanding free software project

pde replied:
I'm fine with either "EFF and others" or "The Certbot project".  We could also make a Contributors file.  Github is almost good enough to make that unnecessary, but not quite &mdash; it misses people who don't have the right emails associated with their accounts.  For instance @schoen has made 681 commits to this repo but isn't in  [the contributor list](https://github.com/letsencrypt/letsencrypt/graphs/contributors).

kuba commented:
This PR will break a lot of PRs in progress. :(

bmw commented:
I know this will break a lot of in progress PRs. This work has to be done though. The client team may be able to help fix up other PRs from open source contributors.

SwartzCr commented on certbot-apache/docs/conf.py:
```diff
@@ -314,5 +314,5 @@
     'python': ('https://docs.python.org/', None),
     'acme': ('https://acme-python.readthedocs.org/en/latest/', None),
-    'letsencrypt': ('https://letsencrypt.readthedocs.org/en/latest/', None),
+    'certbot': ('https://letsencrypt.readthedocs.org/en/latest/', None),
```
we should insert a note to remind us to update this links when the docs move

SwartzCr commented on certbot-apache/setup.py:
```diff
@@ -29,11 +29,11 @@
+    description="Apache plugin for Certbot",
     url='https://github.com/letsencrypt/letsencrypt',
-    author="Let's Encrypt Project",
+    author="Electronic Frontier Foundation",
```
similar authorship note

SwartzCr commented on certbot-compatibility-test/docs/conf.py:
```diff
@@ -311,9 +311,9 @@
     'acme': ('https://acme-python.readthedocs.org/en/latest/', None),
-    'letsencrypt': ('https://letsencrypt.readthedocs.org/en/latest/', None),
-    'letsencrypt-apache': (
+    'certbot': ('https://letsencrypt.readthedocs.org/en/latest/', None),
```
similarly watch this for docs updates

SwartzCr commented on certbot-compatibility-test/setup.py:
```diff
@@ -31,11 +31,11 @@
+    description="Compatibility tests for Certbot",
     url='https://github.com/letsencrypt/letsencrypt',
-    author="Let's Encrypt Project",
+    author="Electronic Frontier Foundation",
```
authors note here too

SwartzCr commented on certbot-compatibility-test/setup.py:
```diff
@@ -31,11 +31,11 @@
+    name='certbot-compatibility-test',
     version=version,
-    description="Compatibility tests for Let's Encrypt client",
+    description="Compatibility tests for Certbot",
```
I think we've agreed to have this be lowercase when not at the start of a sentence

SwartzCr commented on certbot-nginx/README.rst:
```diff
@@ -0,0 +1 @@
+Nginx plugin for Certbot
```
lowercase

SwartzCr commented on certbot/account.py:
```diff
@@ -81,15 +81,15 @@ def __eq__(self, other):
         return
     reporter.add_message(
-        "Your account credentials have been saved in your Let's Encrypt "
+        "Your account credentials have been saved in your Certbot "
```
lower

SwartzCr commented on certbot/account.py:
```diff
@@ -81,15 +81,15 @@ def __eq__(self, other):
         "configuration directory at {0}. You should make a secure backup "
         "of this folder now. This configuration directory will also "
-        "contain certificates and private keys obtained by Let's Encrypt "
+        "contain certificates and private keys obtained by Certbot "
```
lower

SwartzCr commented on certbot/errors.py:
```diff
@@ -1,8 +1,8 @@
 
 class Error(Exception):
-    """Generic Let's Encrypt client error."""
+    """Generic Certbot client error."""
```
lower

SwartzCr commented on certbot/interfaces.py:
```diff
@@ -130,7 +130,7 @@ def more_info():
 
 class IAuthenticator(IPlugin):
-    """Generic Let's Encrypt Authenticator.
+    """Generic Certbot Authenticator.
```
lower

SwartzCr commented on certbot/interfaces.py:
```diff
@@ -230,7 +230,7 @@ class IConfig(zope.interface.Interface):
 
 class IInstaller(IPlugin):
-    """Generic Let's Encrypt Installer Interface.
+    """Generic Certbot Installer Interface.
```
lower

SwartzCr commented on certbot/storage.py:
```diff
@@ -186,9 +186,9 @@ def _is_cli_default(option, value):
-    of the Let's Encrypt client, indicated by the existence of an
-    associated renewal configuration file.
+    Represents a lineage of certificates that is under the management of
+    Certbot, indicated by the existence of an associated renewal
```
lower

SwartzCr commented:
Not sure /all/ of these are relevant, feel free to ignore ones that you think aren't needed

letshelp-certbot/docs/conf.py
62:copyright = u'2014-2015, Let\'s Encrypt Project'
63:author = u'Let\'s Encrypt Project'
246:     u'Let\'s Encrypt Project', 'manual'),

certbot/client.py
54:        ua = "LetsEncryptPythonClient/{0} ({1}) Authenticator/{2} Installer/{3}"

LICENSE.txt
1:Let's Encrypt Python Client

certbot-compatibility-test/certbot_compatibility_test/util.py
1:"""Utility functions for Let"s Encrypt plugin tests."""

certbot-compatibility-test/docs/conf.py
63:copyright = u'2014-2015, Let\'s Encrypt Project'
64:author = u'Let\'s Encrypt Project'
248:     u'Let\'s Encrypt Project', 'manual'),

letsencrypt-auto
189:          echo "Let's Encrypt apache plugin..."
429:    echo "Sorry, I don't know how to bootstrap Let's Encrypt on your operating system!"
826:  echo "Requesting root privileges to run letsencrypt..."
834:  # letsencrypt-auto (which is always the same as that of the letsencrypt
835:  # package). Phase 2 checks the version of the locally installed letsencrypt.
918:    """Return the latest stable release of letsencrypt."""

tests/travis-integration.sh
19:python test/integration-test.py --letsencrypt

tests/letstest/multitester.py
2:Letsencrypt Integration Test Tool
6:- Copies letsencrypt repo and puts it on the instances
7:- Runs letsencrypt tests (bash scripts) on all of these

tests/letstest/scripts/test_leauto_upgrades.sh
6:cd letsencrypt

tests/letstest/scripts/test_letsencrypt_auto_venv_only.sh
5:cd letsencrypt

tests/letstest/scripts/test_tox.sh
4:# The path to the letsencrypt-auto script.  Everything that uses these might
6:LEA_PATH=./letsencrypt/
14:"$LEA_PATH/letsencrypt-auto" --os-packages-only
16:cd letsencrypt

tests/letstest/scripts/test_apache2.sh
25:    sudo mkdir /etc/httpd/sites-available #letsencrypt requires this...
26:    sudo mkdir /etc/httpd/sites-enabled #letsencrypt requires this...
38:# Run letsencrypt-apache2.
39:cd letsencrypt
48:sudo venv/bin/letsencrypt -v --debug --text --agree-dev-preview --agree-tos \

tests/letstest/scripts/test_renew_standalone.sh
6:# run letsencrypt-apache2 via letsencrypt-auto
7:cd letsencrypt

tests/boulder-integration.sh
7:#   SERVER: Passed as "letsencrypt --server" argument.

certbot-apache/certbot_apache/configurator.py
1086:            # Check if LetsEncrypt redirection already exists
1128:        identical to Letsencrypt's redirection rewrite rule.

certbot-apache/docs/conf.py
69:copyright = u'2014-2015, Let\'s Encrypt Project'
70:author = u'Let\'s Encrypt Project'
253:     u'Let\'s Encrypt Project', 'manual'),

Dockerfile-dev
12:# through the CLI (letsencrypt-docker wrapper that uses standalon

bmw commented:
I believe I addressed most review comments. The things I did not do are:

* lowercase Certbot after IRC discussion
* Make references to all readthedocs locations. Instead I opened #2845.
* Modify places where we make references to a 2014 - 2015 copyright from the Let's Encrypt project. Not sure what we want to do there.
* Modify `letsencrypt-auto` in the root directory. The purpose of this file is to remain identical to the version from the last release tag to help people running ancient versions of le-auto.

If there's anything else I missed, please let me know!

SwartzCr commented:
Looks good to me, pulling the switch
*sheds a tear for all unmerged PRs*
